### ISSUE TITLE
KeyError: "name='sid', domain=None, path=None"

### ISSUE THREAD

**Issue by briandaltilio** [Nike-Inc/gimme-aws-creds#324]
New installation of gimme-aws-creds v2.4.0 on Windows 10.  Python version 3.10.0.  After entering Okta credentials, a bunch of errors are displayed ending with KeyError: "name='sid', domain=None, path=None".  

Anonymized output and config file attached.
[okta_aws_login_config.txt](https://github.com/Nike-Inc/gimme-aws-creds/files/7468193/okta_aws_login_config.txt)
[error-output.txt](https://github.com/Nike-Inc/gimme-aws-creds/files/7468194/error-output.txt)

**Comment by PeteW** [comment]
it is a known error in gimme-aws-creds v 2.4.0
you must upgrade to gimme-aws-creds 2.4.3

so you will try to run something like `pip install --upgrade gimme-aws-creds`
and you will see that `gimme-aws-creds --version` yields 2.4.0 and you failed to upgrade

then you may try to run `pip install --upgrade gimme-aws-creds==2.4.3`

and you will get an error message like `ERROR: No matching distribution found for winrt>=1.0.20330.1; platform_system == "Windows" and python_version >= "3.7"`

and you will scratch your head because 3.10 > 3.7 but then your head will hurt because "3.10" < "3.7" 
and then if youre like me you will install python 3.9 and then you will get around this.

anyways this was my afternoon today

**Comment by rsichnyi** [comment]
It has nothing to do with string comparisons in requirements (that issue was fixed in pip back in 2019). The reason is that there's no winrt release for 3.10, see https://github.com/microsoft/xlang/issues/763

**Comment by burrand** [comment]
Any idea how we can work around this? Is it just to downgrade your python version from 3.10 to 3.9?

**Comment by rsichnyi** [comment]
yes, you can only install it on python 3.9 (or lower) until winrt is upgraded to 3.10

**Comment by burrand** [comment]
Thank you for your reply, that worked for me.

**Comment by Loukili-Rachid** [comment]
> Any idea how we can work around this? Is it just to downgrade your python version from 3.10 to 3.9?

https://www.python.org/downloads/release/python-3912/

**Comment by jmoraRMS** [comment]
I already downgrade my python to 3.9 and even with that change I have the same issue. Any help of what I can do?

**Comment by ngalaby-jhb** [comment]
Having same issues with python 3.10 running on Macos. gimme-aws-creds version = 2.4.4

**Comment by santoshgonela** [comment]
I have python 3.9.13 and gimme creds 2.4.4 , when I use the command "gimme-aws-creds" I see my terminal is hanging at "Using password from keyring" and does not respond. Can anyone help me on how to resolve this please?

**Comment by Kyle-Payne** [comment]
Team, per https://github.com/microsoft/xlang/issues/763 are there plans to replace winrt with https://pypi.org/project/winsdk/ in gimme-aws-creds in the near future?

**Comment by kishorer00** [comment]
Finally, Install gimme creds without any error on my system.
Here are the steps ,
==> Install 2.3.1 
#pip install gimme-aws-creds==2.3.1
==> Upgrade to 2.4.3
#pip install --upgrade gimme-aws-creds==2.4.3

It worked for me !!!.
